pkg/daemon/update.go

```diff
@@ -1797,11 +1795,75 @@ func (dn *Daemon) updateSSHKeys(newUsers []ign3types.PasswdUser) error {
 	return nil
 }
 
+// ExtractAndUpdateOS attempts to consolidating various OS update paths to use the same entrypoint here,
+// to move towards handling CoreOS Layering.
+// Does both extraction (old model) and direct updates (new model)
+// Returns the extracted location to remove
+func ExtractAndUpdateOS(newOSImageURL string) (osImageContentDir string, err error) {
+	if isLayeredUpdate(newOSImageURL) {
+		// support new flow here
+		// Not sure if these commands are enough?
+		glog.Infof("New format OS image detected. Doing a rebase directly")
+		args := []string{"rebase", "--experimental", fmt.Sprintf("ostree-unverified-registry:%s", newOSImageURL)}
+		err = runRpmOstree(args...)
+		return
+	} else {
+		if osImageContentDir, err = ExtractOSImage(newOSImageURL); err != nil {
+			return
+		}
+
+		if err = addExtensionsRepo(osImageContentDir); err != nil {
+			return
+		}
```

Comment:
Is there a reason you moved this here? It's unneeded by the pivot and bootstrap paths, correct? And do they clean it up? I only see the cleanup `applyOsChanges`. +1 for removing pivot and bootstrap if possible :wink:

Reply:
I just put it here as a placeholder since we no longer ship the extensions repo in the base OS. I can always move it into a separate function and add a check.

The other paths don't properly do a full OS update so I thought it wasn't really a big issue either way, since if it isn't used, it is just a no-op

Reply:
The "bootstrap" path isn't really bootstrapping either, so I think that behaviour was technically wrong. Again, I think it isn't used, so I can try removing it